### DISCUSSION TITLE
ComboStrap for layout, DokuWiki for content

### DISCUSSION THREAD

**Discussion by vaniwiki** [combostrap/combo#99] [General]
Over the past 10 years, I have added via DokuWiki plugins many enhancements to my website. At one point I was hacking the code of Nico's Bootie theme to insert menus and conditional includes via Andi's tplinc plugin, and had 1200+ lines of additional css.

Since then, Combo(Strap) has added most stuff I need (custom css now reads ~50 lines), and more in the form of FrontMatter with page queries. Having precise control over SEO and meta data is crucial to any CMS, and I think this such additions belong more in DokuWiki core than in a theme or plugin, but still getting this on top of DokuWiki is a game changer.

However, seeing how my website crashed so bad recently, and the many issues and errors I see in my log from third party plugins that I rely on, I would feel better if ComboStrap had a "compatibility mode," letting DokuWiki syntax remain as-is. Or a "safe mode", where it doesn't touch output at all, yet still offering UI components and frontmatter functionality. If that makes sense.

Instead of having to fix all errors on my website, or continue with broken functionality, I'd much rather have the functionality back, and lose some of the ComboStrap finesse, like extra syntax in headings and so on. I believe I can still do that inside a <heading> element? And the same for links, inside a <link> element? Then the heading and link setting would be obsolete for me.

If I'd get back all the functionality that has gone missing over the years (include plugin sections, search highlights from links, and a few more if I think back), I'll happily trade that for some inconvenience or compromise. Turning these setting off leaves the page explorer not working, but I could get around this by making a lot of sidebars in all namespaces with links, as my content is mainly static. Or there could be other simple navigational plugins that fulfils my needs. The automatic tooltips also stops working, but we had another plugin for that in the past, that possibly could be rewritten to fetch the json description. I don't know how much work is there, but the developer was responsive and helpful in the past. And with only one plugin to consider, not a whole collection...

The best case scenario is to get back all the functionality that I had working nicely in the past, plus as much as possible/needed of Combo. But as a humble beggar, it's not up to me. Just throwing it out there for consideration... Thanks anyhow! :)

**Comment by gerardnico** [comment]
It turns out, you can:
https://combostrap.com/docs/theme-b4h63iu5#disable

**Comment by gerardnico** [comment]
I get it that it may be frustrating and I'm sorry for that.

I got all in one place because dokuwiki does not allow easy integration of codes.

How do you add the template field to change it on the fly in a Dokuwiki Template. You don't. You need to go through the hassle of the administration tool and it's for the whole template not by page.

You will never find a plugin that allow you to have a different layout by page.
- with metadata change in a forms
- with full templating engine (not page as dokwuiki does but by fragment)

There is a plugin that allows you to switch from template all together but this is not templating, this is theming because you need to recreate the whole page.

Life is a thread off.

Without taking the link, you would not get any preview. 
Preview is a pretty complicated feature that needs to get the info of the page.

Dokwuiki does not allow to change the AST, it's now pretty complicated but believe me, I would have choose the easy path if I could have.

All the best

**Reply by vaniwiki** [reply]
> You will never find a plugin that allow you to have a different layout by page.

The [tplinc plugin](https://doku.wiki/plugin:tplinc) allows you to define conditions that will display certain templates based on your location. Of course, it is not the same as Combo, but it has existed for quite some time. You can include your own "slots" programatically based on namespace, pages, etc.

> It turns out, you can:
> https://combostrap.com/docs/theme-b4h63iu5#disable

I don't want to turn off the ComboStrap theme like that, I still want to retain the look and feel of Combo. I want Combo to deal with the layout and design, and DokuWiki to spit out content like before. Of course, enhanced by Combo syntax, but not overriding the DokuWiki, at least not irreversible.

There is a lot of functionality that I can't use in ComboStrap, like URL manager (clashes with redirect plugin). I don't use numbered headings. I don't really need any fancy syntax in headings and links, as I am but a simple archivist of sacred scripture.

**Reply by gerardnico** [reply]
This is not templating, this is conditional inclusion.

You don't choose directly a template.

A real template (not dokuwiki template) is going to adapt to the content, therefore it needs to know before hands the slot.

Good luck with that with this plugin as you can't.

**Reply by gerardnico** [reply]
Without taking into account that a toc is also a slot.
Therefore your template css should also take that into account.

Combo does it, tplinc or any dokuwiki template does not because a dokuwiki template drives the rendering while in combo that's the content that does. Ie rendering first, then templating.

**Comment by gerardnico** [comment]
In other words:
- all in with Combo
- all in with Dokuwiki Plugin

With Dokuwiki, I guarantee you that you will never succeed in having the same level of functionality as combo. Why? I tried. That's why combo exists in the first instance.

I created the 404 plugin, the minimap plugin, and then not one but 2 templates (Bootie, then Strap), and every single time, things get falling apart like you experience.

I got tired of that and Combostrap was born.

**Reply by vaniwiki** [reply]
I have been a fan since Bootie, jumping ship from Boostrap3, which got bloated in the end, I felt. I needed simple and fast.

What I want is ComboStrap, but working without the extra headings and links syntax. That should solve the outstanding issues I have I believe. The page exlorer and autotooltip functionality would be sacrificed. Plus there is that strange rendering issue. But I had autotooltip working before via another plugin, so it could be done. And navigation, I can build up manually.

I actually only recently changed all my start pages from being in parent namespace, named like namespace, to being a start page inside the namsapce. So I never had minimap functionality working nicely. But with proper detection of start pages, it works. How come that this relies on the link syntax setting?
